spec/chewy/type/import/bulk_builder_spec.rb

```diff
@@ -175,6 +175,65 @@
         end
       end
     end
+
+    context 'with parents' do
+      let(:type) { CommentsIndex::Comment }
+
+      before do
+        stub_model(:comment)
+        stub_index(:comments) do
+          define_type Comment do
+            field :content
+            #TODO extract `join` type handling to the production chewy code to make it reusable
+            field :join_field, type: :join, relations: {question: :answer}, value: -> { parent.present? ? {name: :answer, parent: parent} : :question }
```

Comment:
Looks like we have to also consider parent document's shard: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/parent-join.html#_parent_join_restrictions
> Parent and child documents must be indexed on the same shard. This means that the same routing value needs to be provided when getting, deleting, or updating a child document. 

Moreover, for multiple levels, we have to consider grand-parent's shard: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/parent-join.html#_multiple_levels_of_parent_join
> Indexing a grandchild document requires a routing value equals to the grand-parent (the greater parent of the lineage)